### PR TITLE
Generic processing logic

### PR DESCRIPTION
**Pull request by minecraft7771**
# Features
- Adds a generic processing logic, which is suitable for most multis to just use. This way it's faster to change stuff in all multis and it potentially reduces bug likelyhood.
- Deprecated `protected boolean CheckRecipe(ItemStack stack)` in favor of `protected CheckRecipeResult checkProcessing()` which will use the generic processing logic to perform recipe check. For new multis this new method should be overriden if the generic logic should not be used.
- Added `FindRecipeResult findRecipeWithResult(...)` in order to get more information why a recipe failed like not enough voltage etc.
- Added many failure messages as to why a multi doesn't start. They will be displayed in the controller GUI. Addons can earily expand the list and add more.
- Implementation is backwards compatible, so no need to migrate all machines immiedietly.
- Close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/13780
- Close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/13869
# Images

### CODE REVIEW

**Review by BlueWeabo — COMMENTED** [review]
Overall looks good! Great job on the both of you.

File: src/main/java/gregtech/api/logic/ProcessingLogic.java [hunk 1]
```diff
@@ -49,35 +99,259 @@ public ProcessingLogic setCurrentOutputFluids(FluidStack... currentOutputFluids)
...
+
+        return CheckRecipeResultRegistry.SUCCESSFUL;
+    }
+
+    /**
+     * Override to tweak final duration that will be set as a result of this logic class.
+     */
+    protected double calculateDuration(@Nonnull GT_Recipe recipe, @Nonnull GT_ParallelHelper helper,
+        @Nonnull GT_OverclockCalculator calculator) {
+        return calculator.getDuration() * helper.getDurationMultiplierDouble();
```
[the remark this answers is not in the document]
Comment (BlueWeabo) [reply src/main/java/gregtech/api/logic/ProcessingLogic.java:311]: Smoll night brain

**Review by BlueWeabo — APPROVED** [review]
Didn't want to approve before i got a second look in the morning.

**Comment by Dream-Master** [comment]
Work for me. I tested it for 2 weeks and all issues i reported are fixed.

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**